pkgs/development/interpreters/python/cpython/default.nix

```diff
@@ -361,6 +361,11 @@ in with passthru; stdenv.mkDerivation {
     # This allows build Python to import host Python's sysconfigdata
     mkdir -p "$out/${sitePackages}"
     ln -s "$out/lib/${libPrefix}/"_sysconfigdata*.py "$out/${sitePackages}/"
+
+    # debug info can't be separated from a static library and would otherwise be
+    # left in place by a separateDebugInfo build. force its removal here to save
+    # space in output.
+    $STRIP -S $out/lib/${libPrefix}/config-*/libpython*.a || true
```

Comment:
How do we deal with debug info in case of static builds? Do we always just remove it?

Aside from this question this PR looks good to me.

Reply:
Various sources suggest that the thing to do is just put the whole debug static lib into the debug output. But doing this gets into the sticky situation of whether a `debug` output actually exists for platforms not supporting `separateDebugInfo`. And rather than doing something awkward that might break obscure platform x and get the whole PR reverted (again) I thought I'd keep this PR as straightforward as possible.